src/pyFAI/gui/pilx/MainWindow.py

```diff
@@ -111,21 +111,32 @@ def __init__(self) -> None:
         self._fixed_indices = set()
         self._background_point = None
 
-    def initData(self, file_name: str):
+    def initData(self,
+                 file_name: str,
+                 dataset_path:str = "/entry_0000/measurement/images_0001",
+                 nxprocess_path:str = "/entry_0000/pyFAI",
+                 ):
+
         self._file_name = os.path.abspath(file_name)
+        self._dataset_path = dataset_path
+        self._nxprocess_path = nxprocess_path
+
         self.sigFileChanged.emit(self._file_name)
 
         with h5py.File(self._file_name, "r") as h5file:
-            map = get_dataset(h5file, "/entry_0000/pyFAI/result/intensity")[:, :, 0]
+            map = get_dataset(h5file, f"{self._nxprocess_path}/result/intensity")[:, :, 0]
```

Comment:
```suggestion
            nxprocess = h5file[self._nxprocess_path]
            map = get_dataset(nxprocess, "/result/intensity")[:, :, 0]
```

And then you can reuse `nxprocess` below